GitCommands/Git/Executable.cs

```diff
@@ -210,12 +207,12 @@ public StreamReader StandardError
             {
                 get
                 {
-                    if (!_redirectOutput && !_throwOnErrorOutput)
+                    if (!_redirectOutput && !_throwOnErrorExit)
                     {
                         throw new InvalidOperationException("Process was not created with redirected output.");
                     }
 
-                    if (!_throwOnErrorOutput)
+                    if (!_throwOnErrorExit)
```

Comment:
Should this check be removed and error be handled as stdout?
Now there is no stderr if no error exit.

Reply:
If there is an error (non-zero exit code), the redirected stderr is used for the exception message.
If there is no error but text on stderr, it will not be evaluated by most of the callers anyway.
In a follow-up, I want to add the stderr output to the Git Command Log. Then I will review which functions evaluate stderr at all.
Perhaps the `StandardError` property can be removed. Or else I will `tee` this stream for logging and for evaluation by the caller.

Reply:
Until the follow up please retain the original behavior, i.e. 
```diff
-                           if (_throwOnErrorOutput)
-                           {
                                string errorOutput = _process.StandardError.ReadToEnd().Trim();
-                               if (exitCode != 0 || errorOutput.Length > 0)
+                               if (exitCode != 0 || (errorOutput.Length > 0 && _throwOnErrorExit))
                                {
                                ExternalOperationException ex
                                    = new(command: _process.StartInfo.FileName,
```

Reply:
This is OK, stderror is evaluated separately.